versioned/tiered/tiered-impl/src/main/java/org/projectnessie/versioned/impl/TieredVersionStore.java

```diff
@@ -435,7 +436,7 @@ public void assign(NamedRef namedRef, Optional<Hash> currentTarget, Hash targetH
   }
 
   @Override
-  public Stream<Key> getKeys(Ref ref) throws ReferenceNotFoundException {
+  public Stream<WithEntityType<Key>> getKeys(Ref ref) throws ReferenceNotFoundException {
```

Comment:
Wouldn't this change back once the filter is pushed down?

Reply:
no, as the type is still pushed to the client